File: LB1/Task1/character/character.py
from LB1.Task1.character.item import Weapon, Armor


class Character:
    def __init__(self, weapon: Weapon, armor: Armor):
        self.weapon = weapon
        self.armor = armor

    def attack(self) -> str:
        return self.weapon.attack()

    def defend(self) -> str:
        return self.armor.defend()

File: LB1/Task3/src/Character.java
import items.Armor;
import items.Weapon;

public class Character {
    private final Weapon weapon;
    private final Armor armor;

    public Character(Weapon weapon, Armor armor) {
        this.weapon = weapon;
        this.armor = armor;
    }

    public void attack() {
        this.weapon.attack();
    }

    public void defend() {
        this.armor.defend();
    }
}
